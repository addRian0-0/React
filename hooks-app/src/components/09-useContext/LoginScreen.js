import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const { user, setuser } = useContext(UserContext);
    console.log(user)

    return (
        <div>
            <h1>Login Screen</h1>
            <hr />
            <button
                onClick={() => setuser({
                    id: 123456,
                    name: 'addRian0_0',
                    email: 'addRian@gmail.com'
                })}
                className="btn btn-primary"
            >
                Login
            </button>
        </div>
    )

}
