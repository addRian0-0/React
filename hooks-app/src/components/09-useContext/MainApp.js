import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'

export const MainApp = () => {

    const [user, setuser] = useState({
        id: 1234,
        name: 'addRian0-0',
        email: 'addrianun@gmail.com'
    });

    return (
        <div>
            <UserContext.Provider value={{ user, setuser }} >
                <AppRouter />
            </UserContext.Provider>
        </div>
    )
}
