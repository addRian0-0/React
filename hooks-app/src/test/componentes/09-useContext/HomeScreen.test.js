import { shallow, mount } from "enzyme";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('pruebas en HomeScreen', () => {

    const user = {
        name: "addRian0-0",
        email: "addrianun@gmail.com"
    }

    const wrapper = mount(
        <UserContext.Provider
            value={{ user }}>
            <HomeScreen
            />
        </UserContext.Provider>
    );

    test('debe de hacer el match', () => {
        expect(wrapper).toMatchSnapshot();
    })


})
