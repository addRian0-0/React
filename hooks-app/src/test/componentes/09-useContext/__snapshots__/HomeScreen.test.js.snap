// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en HomeScreen debe de hacer el match 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
   "name": "addRian0-0",
   "email": "addrianun@gmail.com"
}
  </pre>
</div>
`;
